WordPress/gutenberg
# Off canvas menu editing: adds list view to navigation block

## What?

Part of #42257
Closes #45444

Adds a list view to the navigation block inspector.

## Why?


Using and locating nested items that are hidden in navigation submenus is difficult.

## How?


Copies the implementation in the site editor navigation sidebar which uses a block editor provider to allow the list view to not lose selection to the block in the canvas. 

## Testing Instructions





1. Switch to this PR
2. Go to Gutenberg > Experiments
3. Toggle on "Test a new off canvas editor for navigation block "
4. Save
5. Create a post
6. Add a navigation block
7. Create a block menu with sample items if one is not selected (it has to be a menu)
8. Select the block
9. Notice the new list view in the block inspector
10. Notice the dotted "more menu" in the panel that contains the list view, this is where the menu selector is now.

## Screenshots or screencast 


https://user-images.githubusercontent.com/107534/199258779-af083416-e0c1-4169-b3bd-233a4adfccd3.mp4



## Todo / known issues

- [ ] ~~Add an "Add menu item" item to the end of the list view (asked if this should be a list view item in general)~~
- [ ] Bring back the appender inside the nested branches of the list view
- [x] ~~Fix spacing of manage menus (asked for mockup in design issue)~~
- [x] Move Manage menus to the advanced area
- [x] Cleanup places where list view is not needed (e.g. when there is no menu)
- [x] Find other ways to disable canvas selection via Listview and drop the use of block editor provider.
- [ ] Use [a scrollbar](https://github.com/WordPress/gutenberg/issues/42257#issuecomment-1262499986) on the list view's container only for nested expansions, instead of overflowing the entire inspector.
